modal/stub.py

```diff
@@ -846,10 +848,30 @@ async def interactive_shell(self, cmd=None, image=None, **kwargs):
         async with self.run():
             await wrapped_fn.call(cmd)
 
+    def _make_remote_cls(
+        self, user_cls: type, partial_functions: Dict[str, _PartialFunction], fn_kwargs: Dict[str, Any]
+    ):
+        original_sig = inspect.signature(user_cls.__init__)
+        new_parameters = [param for name, param in original_sig.parameters.items() if name != "self"]
+        sig = inspect.Signature(new_parameters)
+        # TODO: validate signature has only primitive types.
+
+        def __init__(_self, *args, **kwargs):
+            bound_args = sig.bind(*args, **kwargs)
+
+            function_handles: Dict[str, _FunctionHandle] = {}
+
+            for k, v in partial_functions.items():
+                function_handles[k] = self.function(**fn_kwargs, _bound_args=bound_args)(v)
+
+            _PartialFunction.initialize_cls(_self, function_handles)
+
+        return type(f"Remote{user_cls.__name__}", (user_cls,), {"__init__": __init__})
```

Comment:
Feels weird to subclass the original class here. I think we should only copy over the `_PartialFunction` descriptors.

Reply:
yeah i think that makes sense

Reply:
one idea is to define a new base class for all `Remote` classes and inherit from that one. That way you can put the constructor on that class